wrapanapi/systems/lenovo.py

```diff
@@ -453,3 +456,143 @@ def get_device_unique_id(device):
+        """
+        for delegate_to in self._delegate:
+            if hasattr(delegate_to, attr):
+                return getattr(delegate_to, attr)
+        raise AttributeError(attr)
+
+
+class LenovoSwitches(object):
+    """ Encapsulates all Lenovo Switches behavior """
+    API_PATH = "switches"
+
+    _switches_inventory_available = {
+        'product_name': lambda self, requester: self.get_switch_product_name(requester.name),
+        'uuid': lambda self, requester: self.get_switch_uuid(requester.name),
+        'ports': lambda self, requester: self.get_switch_ports(requester.name),
+        'power_state': lambda self, requester: self.get_switch_power_status(requester.name),
+        'health_state:': lambda self, requester: self.get_switch_health_state(requester.name),
+        'type': lambda self, requester: self.get_switch_type(requester.name),
+        'ipv4_addresses': lambda self, requester: self.get_switch_ipv4_addresses(requester.name),
+        'ipv6_addresses': lambda self, requester: self.get_switch_ipv6_addresses(requester.name),
+        'manufacturer': lambda self, requester: self.get_switch_manufacturer(requester.name),
+        'description': lambda self, requester: self.get_switch_description(requester.name),
+        'serial_number': lambda self, requester: self.get_switch_serial_number(requester.name),
+        'part_number': lambda self, requester:
+            self.get_switch_part_number(requester.name),
+        'ipv4_assignments': lambda self, requester:
+            self.get_switch_ipv4_assignments(requester.name),
+        'ipv6_assignments': lambda self, requester:
+            self.get_switch_ipv6_assignments(requester.name),
+        'firmwares': lambda self, requester: self.get_switch_firmwares(requester.name),
+    }
+
+    def __init__(self, service_instance):
+        self._service_instance = service_instance
+
+    def switch_inventory(self, requester=None, *requested_items, **kwargs):
+
+        # Retrieve and return the switch inventory
+        requested_items = requested_items or self._switches_inventory_available
+
+        return {item: self._switches_inventory_available[item](self, requester)
+                for item in requested_items}
+
+    def _list_switches(self):
+        # Collect the switches associated with this LXCA
+        self._switches_list = self._service_instance["switchList"]
+        return self._service_instance["switchList"]
+
+    @property
+    def switches_list(self):
+        if not self._switches_list:
+            self._list_switches()
+        return self._switches_list
+
+    def get_switch(self, switch_name):
+        for switch in self._switches_list:
+            if 'name' in switch and switch['name'] == switch_name:
+                return switch
+        return None
+
+    def get_switch_product_name(self, switch_name):
+        switch = self.get_switch(switch_name)
+        return switch['productName']
```

Comment:
I don't see the value in having so many of these convenience methods that are simply resolving a single key from `self.get_switch`, especially since it isn't even a safe lookup and each of these could generate `KeyError`. Why would the caller use these methods instead of simply calling `LenovoSwitch.get_switch('my_name')['productName']`?